android/src/main/java/com/tailscale/ipn/IPNService.kt

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+package com.tailscale.ipn
+
+import android.app.PendingIntent
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.net.VpnService
+import android.os.Build
+import android.system.OsConstants
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import libtailscale.Libtailscale
+import java.util.UUID
+
+open class IPNService : VpnService(), libtailscale.IPNService {
+  private val randomID: String = UUID.randomUUID().toString()
+
+  override fun id(): String {
+    return randomID
+  }
+
+  override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
+    if (intent != null && ACTION_STOP_VPN == intent.getAction()) {
```

Comment:
intent is always non-null here... and below.

Reply:
the signature implies that intent is non nullable, but [onStartCommand](https://developer.android.com/reference/android/app/Service#onStartCommand(android.content.Intent,%20int,%20int)) can be called with a null intent, like when a service is restarted after being killed by the system and we still have to explicitly check the null case. I think this is just an example of kt's type system not mapped to java's and Android framework methods being defined in Java